miscellaneous/prepare-liferay/prepare.sh

```diff
@@ -11,63 +11,110 @@
+    cd downloads
+
     if [ ! -f "$LIFERAY" ]; then
         echo "... start downloading $LIFERAY (this can take some time)"
-        curl -OskLC - 'https://downloads.sourceforge.net/project/lportal/Liferay%20Portal/6.2.4%20GA5/liferay-portal-tomcat-6.2-ce-ga5-20151119152357409.zip'
+        curl -OsLC - 'https://downloads.sourceforge.net/project/lportal/Liferay%20Portal/7.2.0%20GA1/'"$LIFERAY"
     else
         echo "... the file $LIFERAY already exists and does not need to be downloaded again"
     fi
 
-    LIFERAY_ACTUAL_CHECKSUM=$(md5sum $LIFERAY)
-    if [[ "$LIFERAY_ACTUAL_CHECKSUM" != *"$LIFERAY_CHECKSUM"* ]]; then
+    LIFERAY_ACTUAL_CHECKSUM=$(md5sum $LIFERAY | cut -f 1 -d ' ')
+    if [[ "$LIFERAY_ACTUAL_CHECKSUM" != "$LIFERAY_CHECKSUM" ]]; then
         echo $LIFERAY_ACTUAL_CHECKSUM
         echo "the checksum of $DIR/$LIFERAY does not match"
         echo "the file might be corrput, please remove it and restart $0"
         exit 1
     fi
 
+    echo "...start downloading 3rd party dependencies"
+    mkdir -p modules
+    cd modules
+
+    downloadModule "https://search.maven.org/remotecontent?filepath=commons-codec/commons-codec/1.12/commons-codec-1.12.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=org/apache/commons/commons-collections4/4.1/commons-collections4-4.1.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=org/apache/commons/commons-csv/1.4/commons-csv-1.4.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=commons-io/commons-io/2.6/commons-io-2.6.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=commons-lang/commons-lang/2.4/commons-lang-2.4.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=commons-logging/commons-logging/1.2/commons-logging-1.2.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=com/google/code/gson/gson/2.8.5/gson-2.8.5.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=com/google/guava/guava/21.0/guava-21.0.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=com/fasterxml/jackson/core/jackson-annotations/2.9.8/jackson-annotations-2.9.8.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=com/fasterxml/jackson/core/jackson-core/2.9.8/jackson-core-2.9.8.jar"
+    downloadModule "https://search.maven.org/remotecontent?filepath=com/fasterxml/jackson/core/jackson-databind/2.9.8/jackson-databind-2.9.8.jar"
+
+    cd ../..
+
     echo "... start building $TARGET"
 
-    TMP=$(mktemp -d ${TMPDIR:-/tmp}/tmp.XXXXXXX)
+    TMP="$(mktemp -d ${TMPDIR:-/tmp}/tmp.XXXXXXX)"
+    tar -xzf "downloads/$LIFERAY" -C "$TMP"
+    cp downloads/modules/* "$TMP"/liferay*/osgi/modules/
+    cd "$TMP"
+    mv liferay* sw360
 
-    unzip -q $LIFERAY -d $TMP
-    cd $TMP
-    cp -r liferay*/tomcat* ./sw360
-    rm -r liferay*
+    # Bundle deployables if switched on
+    if [[ $BUNDLE == "1" ]]; then
```

Comment:
I think the bundled result should have a different name or even contain the version of SW360 in the filename. To avoid confusion or unexpected deployment of a different SW360 in the container.

Is this feature even used?

Reply:
This was just a temporary hack to easier create a bundled version. Removed.